### PR TITLE
[NETBEANS-4443] PHP 8.0 Support: Named Arguments

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-4443

- https://wiki.php.net/rfc/named_params

#### Part 3
Don't highlight(as the keyword color) parameter names when they are keywords

e.g.
```php
test(array: [1, 2], default: "default");
```

Notes:

- "default:" is used in a switch statement
- "true:", "false:", and "null:" can be used with a ternary operator
- "else:" is used in alternative syntax for a control structure

nb-php80-named-arguments--reserved-keywords-coloring

#### Part 4

- Support for code completion for parameter names

nb-php80-named-arguments-cc

#### Part 5

- Add whitespace when a parameter name is added via Code Completion

When a parameter name is added via Code Completion, the inserted code is
formatted. However, the whitespace is removed because `functionName(paramName: );` is broken code.

```php

function test($paramName) {}

// before
test(paramName:);

// after
test(paramName: );
```

#### Part 6

- Highlighting for Named Arguments

##### NetBeans

nb-php80-named-arguments-coloring-netbeans

##### FlatLafDark

nb-php80-named-arguments-coloring-dark

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:1437]: Based on this method : https://github.com/apache/netbeans/blob/4e3035d755449849d352439fedc5baf9cfd40ca3/php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java#L1289

**Comment by junichi11** [comment]
@tmysik 
Feature freeze is 18:00 CET (Friday 16 April). https://lists.apache.org/thread.html/rb1b2865c1ca6bd962618b79b2e0c97e0d134a03a05fbfe14a2823c20%40%3Cdev.netbeans.apache.org%3E

So, I think this is too late (and sorry for the huge changes), but could you please review it if you have time? If you don't have time, I'll change the milestone to 12.5, then let's merge this later. (Need not hurry :) It's all my fault.) 
Thanks.

**Comment by geertjanw** [comment]
Looks like great changes but indeed let's change milestone to 12.5.

**Comment by junichi11** [comment]
@geertjanw OK. Thanks!

**Comment by junichi11** [comment]
@geertjanw We can merge it into the `master` branch after the `delivery` branch is created, right?

@tmysik Let's merge it later because the milestone is changed to 12.5 :)

**Comment by geertjanw** [comment]
@junichi11, after the delivery branch is created, we can merge into that branch for inclusion in 12.4, or into master for inclusion in 12.5.

**Comment by geertjanw** [comment]
If @tmysik reviews this in time, I think we should try to include this in 12.4, if you agree.

**Comment by junichi11** [comment]
@geertjanw I see. Thank you.

**Comment by neilcsmith-net** [comment]
>  after the delivery branch is created, we can merge into that branch for inclusion in 12.4,

IMO, it might be better to merge before freeze, review and get testing in the first RC than drop it in later?  Can always revert or tidy up by RC2 if necessary.

**Comment by geertjanw** [comment]
Sure, that would be ideal.

**Review by tmysik — APPROVED** [review]
@junichi11 

The changes are mainly in the tests - that is good; changes in the sources look good to me.
Thanks Junichi for your great work, as always!

**Comment by tmysik** [comment]
@junichi11 @geertjanw @neilcsmith-net 

Guys, approved, see my previous comment. Let me know if we can merge it now or even better, do it, if we can (I maybe won't be available for the rest of the day).

Thanks Junichi for your amazing work.

**Comment by junichi11** [comment]
@tmysik Thanks a lot for your review and your time!

@geertjanw, @neilcsmith-net 

I follow the release managers. If it's OK for you, please feel free to merge it :) Otherwise, let's merge it in 12.5.

**Review by geertjanw — APPROVED** [review]

**Comment by martinm79** [comment]
Hello, I have Netbeans 12.0 but there is no PHP 8 in the selection.
How do I get PHP 8 in Netbeans?

Maybe someone can help me

**Comment by junichi11** [comment]
@martinm79 
NetBeans 12.0 doesn't provide support for PHP 8.0.
Please use the latest version: https://netbeans.apache.org/download/nb124/nb124.html

**Pull request merged**
